src/board/king.rs

```diff
@@ -63,16 +63,14 @@ mod test {
             for file in 0..2 {
                 if (rank, file) != (1, 1) {
                     assert!(moved_boards.iter().any(|x| matches!(
-                        x.get_piece_at_position(Square {
+                        x.get_piece_at_position(UncheckedSquare {
```

Comment:
We know for sure that the squares here are valid, so I don't think we need to validate here.